examples/profile.js

```diff
@@ -25,22 +25,58 @@ app.MainController = function($http, $scope) {
     this['profileData'] = resp.data['profile'];
   }));
 
-  this['point'] = null;
-};
-
-
-/**
- * @param {Object} point The point object. It correspond to an item in the list
- * of points given to the profile.
- */
-app.MainController.prototype.onProfileHover = function(point) {
-  this['point'] = point;
-};
-
-
-/**
- */
-app.MainController.prototype.onProfileOut = function() {
+  /**
+   * @param {Object} item
+   * @return {number}
+   */
+  var z = function(item) {
+    return item['values']['mnt'];
+  };
+
+  /**
+    * @param {Object} item
+    * @return {number}
+    */
+  var dist = function(item) {
+    return item['dist'];
+  };
+
+  /**
+   * @type {ngeox.profile.ProfileExtractor}
+   */
+  var extractor = {z: z, dist: dist};
+
+
+  // Using closures for hoverCallback and outCallback since
+  // wrapping in angular.bind leads to a closure error.
+  var that = this;
+
+  /**
+   * @param {Object} point
+   */
+  var hoverCallback = function(point) {
+    // An item in the list of points given to the profile.
+    that['point'] = point;
+  };
+
+  var outCallback = function() {
+    that['point'] = null;
+  };
+
+
+  /**
+   * @type {ngeox.profile.ProfileOptions}
+   */
+  var profileOptions = {
+    extractor: extractor,
+    hoverCallback: hoverCallback,
+    light: false, // explicitely set to avoid closure error
+    zKeys: undefined, // explicitely set to avoid closure error
```

Comment:
On Tue, Mar 3, 2015 at 9:24 AM, gberaudo <notifications@github.com> wrote:
>
> In examples/profile.js:
>
> > +    that['point'] = point;
> > +  };
> > +
> > +  var outCallback = function() {
> > +    that['point'] = null;
> > +  };
> > +
> > +
> > +  /**
> > +   * @type {ngeox.profile.ProfileOptions}
> > +   */
> > +  var profileOptions = {
> > +    extractor: extractor,
> > +    hoverCallback: hoverCallback,
> > +    light: false, // explicitely set to avoid closure error
> > +    zKeys: undefined, // explicitely set to avoid closure error
>
> Indeed. @elemoine do you know of a better handling of this issue?
> This case is similar to option typedefs we pass to constructors. Ex:
> https://github.com/camptocamp/ngeo/blob/master/examples/geolocation.js#L36
>
> var map = new ol.Map({
>   layers: [
>     new ol.layer.Tile({
>       source: new ol.source.OSM()
>     })
>   ],
>   view: view
> });


You can use a typecast:


var profileOptions = /** @type {ngeox.profile.ProfileOptions} */ ({
  extractor: extractor,
  hoverCallback: hoverCallback
});


-- 
Eric Lemoine

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 94
Mail : eric.lemoine@camptocamp.com
http://www.camptocamp.com